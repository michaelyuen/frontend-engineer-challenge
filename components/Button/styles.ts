import styled from "styled-components";
import { CountdownContainer } from "../Countdown";
import { FormContainer } from "../Form";

// TODO: get button styles from theme
export const ButtonContainer = styled.button`
  color: #ffffff;
  background: #0d6cf2;
  border-radius: var(--border-radius);
  border: 1px solid #0d6cf2;

  &:focus {
    border-radius: var(--border-radius);
    box-shadow: 0 0 0 3px #58a6ff;
    outline: none;
  }

  &[disabled] {
    border-color: grey;
    background: grey;
    cursor: not-allowed;
  }

  ${FormContainer} &, ${CountdownContainer} & {
    align-self: center;
    padding: 1em;
  }
`;
