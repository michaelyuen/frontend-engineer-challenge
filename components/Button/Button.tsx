import { ButtonHTMLAttributes } from "react";
import { ButtonContainer } from "./styles";

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const Button: React.FC<ButtonProps> = ({
  children,
  type = "button",
  ...props
}) => {
  return (
    <ButtonContainer type={type} {...props}>
      {children}
    </ButtonContainer>
  );
};
