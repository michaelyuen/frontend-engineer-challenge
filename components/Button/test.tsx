import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Button } from "./Button";

describe("Button", () => {
  test("renders", () => {
    render(<Button>click me</Button>);
    expect(screen.getByText("click me")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Button>click me</Button>);
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(<Button>click me</Button>);
    expect(await axe(container)).toHaveNoViolations();
  });
});
