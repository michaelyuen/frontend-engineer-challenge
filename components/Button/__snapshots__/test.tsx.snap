// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="sc-gKPRtg cWBLUo"
    type="button"
  >
    click me
  </button>
</div>
`;
